src/pip/_internal/req/req_install.py

```diff
@@ -824,14 +825,38 @@ def install(
             self.install_succeeded = True
             return
 
-        install_legacy(
-            self,
-            install_options=install_options,
-            global_options=global_options,
-            root=root,
-            home=home,
-            prefix=prefix,
-            use_user_site=use_user_site,
-            pycompile=pycompile,
-            scheme=scheme,
-        )
+        # TODO: Why don't we do this for editable installs?
```

Comment:
Why do we do this for non-editable installs? ;)

Reply:
I have no idea :-( It was in the existing code so I just moved it as part of the refactoring. The TODO was to note that it's odd that we do different things for the 2 cases, and now that the code got moved, that difference is more obvious. But fixing it should be a separate PR.